addons/stock/wizard/stock_replenishment_info.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from json import dumps
+
+from odoo import api, fields, models, SUPERUSER_ID
+from odoo.osv.expression import AND
+from odoo.tools.date_utils import get_month, subtract
+
+class StockReplenishmentInfo(models.TransientModel):
+    _name = 'stock.replenishment.info'
+    _description = 'Stock supplier replenishment information'
+    _rec_name = 'orderpoint_id'
+
+    orderpoint_id = fields.Many2one('stock.warehouse.orderpoint')
+    product_id = fields.Many2one('product.product', related='orderpoint_id.product_id')
+    qty_to_order = fields.Float(related='orderpoint_id.qty_to_order')
+    json_lead_days = fields.Char(compute='_compute_json_lead_days')
+    json_replenishment_history = fields.Char(compute='_compute_json_replenishment_history')
+
+    @api.depends('orderpoint_id')
+    def _compute_json_lead_days(self):
+        if not self.orderpoint_id.product_id or not self.orderpoint_id.location_id:
+            return
+        dummy, lead_days_description = self.orderpoint_id.rule_ids._get_lead_days(
+            self.orderpoint_id.product_id, self.orderpoint_id)
+        self.json_lead_days = dumps({
+            'template': 'stock.leadDaysPopOver',
+            'lead_days_date': fields.Date.to_string(self.orderpoint_id.lead_days_date),
+            'lead_days_description': lead_days_description,
+            'today': fields.Date.to_string(fields.Date.today()),
+            'trigger': self.orderpoint_id.trigger,
+            'qty_forecast': self.orderpoint_id.qty_forecast,
+            'qty_to_order': self.orderpoint_id.qty_to_order,
+            'product_min_qty': self.orderpoint_id.product_min_qty,
+            'product_max_qty': self.orderpoint_id.product_max_qty,
+            'product_uom_name': self.orderpoint_id.product_uom_name,
+            'virtual': self.orderpoint_id.trigger == 'manual' and self.orderpoint_id.create_uid.id == SUPERUSER_ID,
+        })
+
+    @api.depends('orderpoint_id')
+    def _compute_json_replenishment_history(self):
```

Comment:
Shouldn't be loop. Isn't It dangerous to do a compute method which accept single record?